tests/framework/matcher/conditions.go

```diff
@@ -107,10 +117,31 @@ func (c conditionMatcher) Match(actual interface{}) (success bool, err error) {
 		return false, nil
 	}
 
-	if c.expectedMessage == "" {
-		return true, nil
+	if c.expectedMessage != "" {
+		return c.checkMessage(conditions)
+	}
+
+	if c.expectedReason != "" {
+		return c.checkReason(conditions)
+	}
```

Comment:
This will not allow to assert a message and a reason at the same time?

Reply:
Correct, if someone wants that use case covered they can extend this later but for now I only need to assert that the reason is set.